### PR TITLE
feat: Added priority queue implementation and its testing. Added message queue implementation.

### PR DESCRIPTION
**Pull request by puchkovki**

### CODE REVIEW

**Review by trmigor — CHANGES_REQUESTED** [review]
Commit 9c779a2e15a38c4a3ef5942a55ce0494fb605619 should be removed.

File: model/MessageQueue.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package model
+
+import (
+	"container/heap"
+	"sync"
+
+	"github.com/puchkovki/distr-model/model"
+	"github.com/puchkovki/distr-model/priorityq"
```
Comment (trmigor) [inline model/MessageQueue.go]: Change puchkovki to trmigor.
Delete "github.com/puchkovki/distr-model/model"

File: priorityq/priorityq_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package priorityq
+
+import (
+	"container/heap"
+	"testing"
+)
+
+func TestPriorityQ(t *testing.T) {
+	// Some items and their priorities.
+	items := map[interface{}]int{
```
Comment (trmigor) [inline priorityq/priorityq_test.go:10]: Better be map[string]int.
But ok, let it be interface{}

File: model/MessageQueue.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+// A MessageQueue is a PriorityQueue structure with mutex for parallel usage.
+type MessageQueue struct {
+	queue priorityq.PriorityQueue
+	mutex sync.Mutex
+}
+
+// Initialiaze establishes the heap invariants.
+func (mq *MessageQueue) Initialiaze() {
```
Comment (trmigor) [inline model/MessageQueue.go]: It should be a constructor. Smth like that:

```
func NewMessageQueue() (mq MessageQueue) {
	heap.Init(&mq.queue)
}
```

File: model/MessageQueue.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	heap.Init(&mq.queue)
+}
+
+// Dequeue removes and returns the object of the priority queue with the minimum priority.
+func (mq *MessageQueue) Dequeue() (ret model.Message) {
+	mq.mutex.Lock()
+	defer mq.mutex.Unlock()
+
+	// Следует ли отлавливать ошибку вида: доступ к пустой очереди?
```
Comment (trmigor) [inline model/MessageQueue.go]: No Russian.
And let the answer be "No".

**Review by trmigor — CHANGES_REQUESTED** [review]

File: model/MessageQueue.go (hunk 1 above)
Comment (puchkovki) [reply model/MessageQueue.go]: How? Your repo doesn't include `priorityq` package.

Comment (trmigor) [reply model/MessageQueue.go]: It will. After I merge PR.

File: model/MessageQueue.go (hunk 4 above)
Comment (puchkovki) [reply model/MessageQueue.go]: Fixed

File: priorityq/priorityq_test.go (hunk 2 above)
Comment (puchkovki) [reply priorityq/priorityq_test.go:10]: Fixed

File: model/MessageQueue.go (hunk 3 above)
Comment (puchkovki) [reply model/MessageQueue.go]: Fixed

**Review by trmigor — APPROVED** [review]
Well done.

**Pull request merged**